src/simple_lottery/proof.rs

```diff
@@ -48,7 +50,21 @@ impl<E: AsRef<[u8]> + Clone, H: Digest + FixedOutput> Proof<E, H> {
                 element_sequence.push(element.clone());
             }
             if element_sequence.len() as u64 >= params.proof_size {
-                element_sequence.sort_unstable_by(|a: &E, b: &E| a.as_ref().cmp(b.as_ref()));
+                let all_have_index = element_sequence.iter().all(|e| e.index.is_some());
```

Comment:
You can replace line 53 to 67 with:
```
element_sequence.sort_unstable_by(|a, b| {
                    a.as_ref().cmp(b.as_ref()).then(
                        if let (Some(a_i), Some(b_i)) = (a.index, b.index) {
                            match a_i.cmp(&b_i) {
                                std::cmp::Ordering::Equal => panic!("a = b, a.index = b.index"),
                                c => c
                            }
                        } else {
                            panic!("a=b, and at least one has no index");
                        },
                    )
                });
 ```
 We will replace the panics with an errors when Result is merge